internal/app/proxyd/internal/frontend/frontend.go

```diff
@@ -86,24 +74,34 @@ func (r *ReverseProxy) Listen(address string) (err error) {
 func (r *ReverseProxy) AddBackend(uid, addr string) (added bool) {
 	r.mux.Lock()
 	defer r.mux.Unlock()
+
+	// Check if we already have this backend added
 	if _, ok := r.backends[uid]; ok {
-		added = false
-		return
+		if r.backends[uid].Addr == addr {
+			added = false
+			return
+		}
 	}
+
+	// Add new backend
 	r.backends[uid] = &backend.Backend{UID: uid, Addr: addr, Connections: 0}
 	added = true
+
+	// Update current backend
 	r.setCurrent()
 
 	return added
 }
 
 // DeleteBackend deletes a backend.
-func (r *ReverseProxy) DeleteBackend(uid string) (deleted bool) {
+func (r *ReverseProxy) DeleteBackend(uid, addr string) (deleted bool) {
 	r.mux.Lock()
 	defer r.mux.Unlock()
 	if _, ok := r.backends[uid]; ok {
-		delete(r.backends, uid)
-		deleted = true
+		if r.backends[uid].Addr == addr {
```

Comment:
I'm not sure this is what we want. We don't care what the IP is.

Reply:
If the uid is reused for whatever reason but has a new IP this would hopefully prevent us from accidentally removing it.